sdl_android_lib/src/com/smartdevicelink/transport/SdlBroadcastReceiver.java

```diff
@@ -203,7 +203,12 @@ private static boolean isRouterServiceRunning(Context context, boolean pingServi
 	            	Intent intent = new Intent();
 	            	intent.setClassName(service.service.getPackageName(), service.service.getClassName());
 	            	intent.putExtra(TransportConstants.PING_ROUTER_SERVICE_EXTRA, pingService);
-	            	context.startService(intent);
+	            	try {
+                        context.startService(intent);
+					}catch (SecurityException e){
+						Log.e(TAG, "Security exception, process is bad");
+						return false; // Let's exit, we can't start the service
```

Comment:
So we actually can't just return false blindly here. If you take a look at the return statement after this for loop you can see we need to actually return something meaningful. I'd move the ping code into a separate method with the try/catch so that we can continue through the loop and return something meaningful. 

Something like this:
```java 
	protected void pingRouterService(Context context,String packageName, String className){
		if(context == null || packageName == null || className == null){
			return;
		}
		try{
			Intent intent= new Intent();
			intent.setClassName(packageName, className);
			intent.putExtra(TransportConstants.PING_ROUTER_SERVICE_EXTRA, true);
    		context.startService(intent);
		}catch(SecurityException e){
			
		}
	}
```